### ISSUE TITLE
DepthCloud memory leak

### ISSUE THREAD

**Issue by christian-rauch** [ros-visualization/rviz#1621]
I am using the `DepthCloud` display type to back-project colour and depth to a 3D point cloud in RViz. When I click the `Add` button I can add a new display type. While this new pop-up window is shown, the point cloud in the 3D view is not updated and the memory consumption by RViz rises until its process gets killed by `oom-kill`.

Edit:
I just realised that this issue has already been reported as https://github.com/ros-visualization/rviz/issues/868.

Reproduce:
1. play a bag file with depth and colour images
2. open RViz
3. add and configure the `DepthCloud` display type
4. try to add a new display type by clicking on the `Add` button

My configuration:
![rviz_screenshot_2021_04_15-15_38_36](https://user-images.githubusercontent.com/8226248/114888100-db37fc00-9e00-11eb-8273-cc1c7a561619.png)

Memeory consumption of the process:
![Screenshot from 2021-04-15 15-27-46](https://user-images.githubusercontent.com/8226248/114888604-4e417280-9e01-11eb-9568-3840579545c2.png)

### Your environment
* OS Version: Ubuntu 20.04
* ROS Distro: Noetic
* RViz, Qt, OGRE, OpenGl version as printed by rviz:
```
rviz version 1.14.5
compiled against Qt version 5.12.8
compiled against OGRE version 1.9.0 (Ghadamon)
Forcing OpenGl version 0.
Stereo is NOT SUPPORTED
OpenGL device: Mesa Intel(R) UHD Graphics 630 (CFL GT2)
OpenGl version: 4.6 (GLSL 4.6) limited to GLSL 1.4 on Mesa system.
```

**Comment by rhaschke** [comment]
This is related to #1606. Stopping data processing as introduced in #631 will result in the accumulation of ROS messages in the callback queue. However, this should only happen for large queue sizes (and yours is 1)!?
https://github.com/ros-visualization/rviz/issues/868#issuecomment-120614852 suggests that there is another cause for that issue.

**Comment by rhaschke** [comment]
Fixed via #1635
